llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -969,6 +969,41 @@ static void translateBlockStore(CallInst &CI, bool IsSLM) {
   SI->setDebugLoc(CI.getDebugLoc());
 }
 
+static void translateGatherLoad(CallInst &CI, bool IsSLM) {
+  IRBuilder<> Builder(&CI);
+
+  constexpr int AlignmentTemplateArgIdx = 2;
+  APInt Val = parseTemplateArg(CI, AlignmentTemplateArgIdx,
+                               ESIMDIntrinDesc::GenXArgConversion::TO_I64);
+  Align AlignValue(Val.getZExtValue());
+
+  auto OffsetsOp = CI.getArgOperand(0);
+  auto MaskOp = CI.getArgOperand(1);
+  auto PassThroughOp = CI.getArgOperand(2);
+  auto DataType = CI.getType();
+
+  // Convert the mask from <N x i16> to <N x i1>.
+  auto NumElems = (cast<VectorType>(DataType))->getElementCount();
+  auto MaskType = VectorType::get(Builder.getInt1Ty(), NumElems);
+  MaskOp = Builder.CreateTruncOrBitCast(MaskOp, MaskType);
```

Comment:
auto-review: This is probably not correct. esimd masks considered set if any bit is set, not the lowest one only. Thus need to use different operation here.